src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsCard.component.ts

```diff
@@ -135,14 +159,36 @@ export class NotebookViewsCardComponent implements OnInit {
 	}
 
 	public get display(): boolean {
-		if (!this._metadata || !this._activeView) {
+		if (!this.activeView) {
 			return true;
 		}
 
-		return !this.data?.hidden;
+		if (!this._metadata) { //Means not initialized
+			return false;
+		}
+
+		return !!this.activeView.displayedCells.find(c => c.cellGuid === this.cell.cellGuid);
+	}
+
+	public get awaitingInput(): boolean {
+		return this.cell.future && this.cell.future.inProgress;
 	}
 
 	public get showActionBar(): boolean {
 		return this.cell.active;
 	}
 }
+
+registerThemingParticipant((theme: IColorTheme, collector: ICssStyleCollector) => {
+	const cellBorderColor = theme.getColor(cellBorder);
+	if (cellBorderColor) {
+		collector.addRule(`.notebookEditor .notebook-cell.active .actionbar { border-color: ${cellBorderColor};}`);
```

Comment:
These seem to be affecting the toolbar for all notebooks, is that intended? Color changes in this file should generally always be scoped down to just notebook views cards. @chlafreniere

Reply:
I refined the scope of the rule so it only applies to the Views